### PR TITLE
Add warning to inventory for settings for cloud connector

### PR DESCRIPTION
**Pull request by MariaAga**
Will need to use the selector from https://github.com/theforeman/foreman_rh_cloud/pull/458 in order to work
Screenshot from 2021-01-27 15-27-57

### CODE REVIEW

**Comment by ares** [comment]
The screenshot looks good

File: webpack/ForemanInventoryUpload/Components/PageHeader/PageHeader.js [hunk 1]
```diff
@@ -4,12 +4,14 @@ import InventorySettings from '../InventorySettings';
...
 import InventoryFilter from '../InventoryFilter';
 import ToolbarButtons from './components/ToolbarButtons';
+import SettingsWarning from './components/SettingsWarning';
 import PageTitle from './PageTitle';
 import './PageHeader.scss';
 
 const PageHeader = () => (
   <div className="inventory-upload-header">
     <PageTitle />
+    <SettingsWarning />
```
Comment (laviro) [inline webpack/ForemanInventoryUpload/Components/PageHeader/PageHeader.js:14]: Can the warnings move above the PageTitle?

File: webpack/ForemanInventoryUpload/Components/PageHeader/components/SettingsWarning/index.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  selectAutoUploadEnabled,
+  selectHostObfuscationEnabled,
+} from '../../../AccountList/AccountListSelectors';
+
+const ConnectedSettingsWarning = () => {
+  const autoUpload = useSelector(selectAutoUploadEnabled);
+  const hostObfuscation = useSelector(selectHostObfuscationEnabled);
+  // TODO: add the selector
+  // const isCloudConnector = useSelector(selectCloudConnector);
+  const isCloudConnector = false;
```
Comment (laviro) [inline webpack/ForemanInventoryUpload/Components/PageHeader/components/SettingsWarning/index.js:14]: now that cloud connector initial PR is merged, can you use its status selector?

File: webpack/ForemanInventoryUpload/Components/PageHeader/components/SettingsWarning/SettingsWarning.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        title={__(
+          "Cloud Connector has been configured however obfuscating host names setting is enabled, it's recommended to disable it"
+        )}
+      />
+    );
+  }
+  if (!autoUpload || hostObfuscation) {
+    return <div className="settings-alert">{alerts}</div>;
+  }
+  return null;
```
Comment (laviro) [inline webpack/ForemanInventoryUpload/Components/PageHeader/components/SettingsWarning/SettingsWarning.js:41]: can you add the opposite condition to line 12 in order to `return null` early?
```suggestion
    return <div className="settings-alert">{alerts}</div>;
```

**Comment by MariaAga** [comment]
Resolved comments and moved the warning to be above the title 
Screenshot from 2021-02-07 13-47-22

**Review by laviro — COMMENTED** [review]
Thanks Maria!
can you also add a closing icon to the alerts?
and also I think there is a small diff between the triangle icon's line-height to the text line-height:

Screenshot (59)

File: webpack/ForemanInventoryUpload/Components/PageHeader/components/SettingsWarning/SettingsWarning.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+export const SettingsWarning = ({
+  autoUpload,
+  hostObfuscation,
+  isCloudConnector,
+}) => {
+  if (!isCloudConnector) {
+    return null;
+  }
+  if (autoUpload && !hostObfuscation) {
```
[the remark this answers is not in the document]
Comment (MariaAga) [reply webpack/ForemanInventoryUpload/Components/PageHeader/components/SettingsWarning/SettingsWarning.js:14]: I prefer this way so the condition is easy to understand

**Comment by MariaAga** [comment]
Screenshot from 2021-02-15 16-24-08
Added Some css, and a close button

File: webpack/ForemanInventoryUpload/Components/PageHeader/components/SettingsWarning/SettingsWarning.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+export const SettingsWarning = ({
+  autoUpload,
+  hostObfuscation,
+  isCloudConnector,
+}) => {
+  const [showUploadWarning, setShowUploadWarning] = useState(true);
+  const [showObfuscationWarning, setShowObfuscationWarning] = useState(true);
+  if (
+    !isCloudConnector ||
+    (!setShowUploadWarning && !setShowObfuscationWarning)
```
Comment (laviro) [inline webpack/ForemanInventoryUpload/Components/PageHeader/components/SettingsWarning/SettingsWarning.js:15]: just wondering, is there a chance the sets won't exist in this line?

**Review by laviro — COMMENTED** [review]
Thanks @MariaAga! works great,
just a small nitpick

File: webpack/ForemanInventoryUpload/Components/PageHeader/components/SettingsWarning/SettingsWarning.js (hunk 5 above)
Comment (MariaAga) [reply webpack/ForemanInventoryUpload/Components/PageHeader/components/SettingsWarning/SettingsWarning.js]: It was suppose to be `showObfuscationWarning` and not `setShowObfuscationWarning`, fixed, thanks!

**Review by laviro — APPROVED** [review]

**Comment by laviro** [comment]
Thanks @MariaAga !

**Pull request merged**